### ISSUE TITLE
istio policies are not cleaned up when kubernetes namespace deleted 

### ISSUE THREAD

**Issue by LillyWu** [istio/istio#6727]
**Describe the bug**
When kubernetes namespaces deleted, istio policies still exists. 

**Expected behavior**
policies should be also cleaned up together with the namespace

**Steps to reproduce the bug**
1. Follow the task in the official doc here:  https://istio.io/docs/tasks/security/authn-policy/
2. clean up all resources with `kubectl delete ns foo bar legacy`
3. check the policies and destinationrules
```
$ istioctl get policy
NAME        KIND                                      NAMESPACE
example-2   Policy.authentication.istio.io.v1alpha1   bar
example-1   Policy.authentication.istio.io.v1alpha1   foo
$ istioctl get destinationrule
NAME              KIND                                           NAMESPACE
istio-policy      DestinationRule.networking.istio.io.v1alpha3   istio-system
istio-telemetry   DestinationRule.networking.istio.io.v1alpha3   istio-system
```
we can see destinationrules have been deleted, but policy not 

**Version**
What version of istio and Kubernetes are you using? Use `istioctl version` and `kubectl version`
```
$ istioctl version
Version: 0.8.0
GitRevision: 6f9f420f0c7119ff4fa6a1966a6f6d89b1b4db84
User: root@48d5ddfd72da
Hub: docker.io/istio
GolangVersion: go1.10.1
BuildStatus: Clean
```
and 
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.5", GitCommit:"32ac1c9073b132b8ba18aa830f46b77dcceb0723", GitTreeState:"clean", BuildDate:"2018-06-21T11:46:00Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9+", GitVersion:"v1.9.8-2+af27ab4b096122", GitCommit:"af27ab4b096122049e65b75ee29ac115b1d58f6b", GitTreeState:"clean", BuildDate:"2018-06-14T04:07:19Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Is Istio Auth enabled or not?**
I installed istio with `istio-0.8.0/install/kubernetes/istio-demo.yaml`
and enable authentication with policies which is required in the task: https://istio.io/docs/tasks/security/authn-policy/

**Environment**
IKS

**Comment by quanjielin** [comment]
didn't get repro in gke env.

**Comment by LillyWu** [comment]
@quanjielin  I am using [IBM Kubernetes Service](https://console.bluemix.net/catalog/?category=containers)

**Comment by irisdingbj** [comment]
@LillyWu  After you run 'kubectl delete ns foo bar legacy' you need to wait for a while to let the namespace get deleted clean. You can double confirm via 'kubectl get namespaces' to make sure all namespaces disappears.  When it is still in 'Terminating' status you will see the policy still there via 'istioctl get policy' . But when the namespace has been deleted, the policy will be deleted as well.  So 
In summary, this can not be reproduced in IBM Kubernetes Service as well.

**Comment by ymesika** [comment]
@irisdingbj Thanks for testing it on IKS.
@LillyWu Please re-open if it indeed doesn't clean up eventually.

**Comment by LillyWu** [comment]
@irisdingbj Thank you for your reply. And I re-tried it. The policies are deleted as expected.
